server/src/main/java/io/druid/server/coordinator/rules/Rules.java

```diff
@@ -27,13 +27,13 @@
 {
   public static boolean eligibleForLoad(Interval src, Interval target)
   {
-    return src.contains(target);
+    return src.overlaps(target);
   }
 
   public static boolean eligibleForLoad(Period period, Interval interval, DateTime referenceTimestamp)
   {
     final Interval currInterval = new Interval(period, referenceTimestamp);
-    return currInterval.overlaps(interval) && interval.getStartMillis() >= currInterval.getStartMillis();
```

Comment:
Does anyone remember or know why `&& interval.getStartMillis() >= currInterval.getStartMillis()` was here in the first place? I can't think of a good reason for it - probably you'd want a load rule to apply if it overlaps a segment all, even if the segment doesn't start within the load rule interval.